editoast/src/views/infra/routes.rs

```diff
@@ -249,57 +280,114 @@ mod tests {
+                ),
             ),
             (
                 json!({ "PH1": null }), // all routes crossing PH1
-                vec![
-                    "rt.DG2->DH1",
-                    "rt.DH2->DG4",
-                    "rt.DH2->buffer_stop.7",
-                    "rt.DH3->DH1",
-                ],
+                (
+                    vec![
+                        "rt.DG2->DH1",
+                        "rt.DH2->DG4",
+                        "rt.DH2->buffer_stop.7",
+                        "rt.DH3->DH1",
+                    ],
+                    vec![(
+                        "PH1".to_string(),
+                        vec!["A_B1".to_string(), "A_B2".to_string()],
+                    )],
+                ),
             ),
             (
                 json!({ "PA0": "A_B1", "PA2": "A_B1" }),
-                vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                (
+                    vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                    vec![
+                        ("PA0".to_string(), vec!["A_B1".to_string()]),
+                        ("PA2".to_string(), vec!["A_B1".to_string()]),
+                    ],
+                ),
             ),
             (
                 json!({ "PA0": "A_B1", "PA2": null }),
-                vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                (
+                    vec!["rt.DA0->DA5", "rt.DA3->buffer_stop.1"],
+                    vec![
+                        ("PA0".to_string(), vec!["A_B1".to_string()]),
+                        ("PA2".to_string(), vec!["A_B1".to_string()]),
+                    ],
+                ),
             ),
         ];
 
         let app = create_test_service().await;
         let small_infra = small_infra(db_pool()).await;
 
-        for (params, mut expected) in tests {
+        fn compare_result(got: RoutesFromNodesPositions, expected: RoutesFromNodesPositions) {
+            let mut got_routes = got.routes;
+            let mut expected_routes = expected.routes;
+            got_routes.sort();
+            expected_routes.sort();
+            assert_eq!(got_routes, expected_routes);
+
+            let process_available_positions = |positions: &HashMap<String, Vec<String>>| {
+                positions
+                    .iter()
+                    .map(|(key, value)| {
+                        let mut sorted_node_positions = value.clone();
+                        sorted_node_positions.sort();
+                        (key.clone(), sorted_node_positions)
+                    })
+                    .collect::<Vec<_>>()
+            };
+
+            let mut got_available_positions: Vec<_> =
+                process_available_positions(&got.available_node_positions);
+            let mut expected_available_positions: Vec<_> =
+                process_available_positions(&expected.available_node_positions);
+
+            got_available_positions.sort();
```

Comment:
You can sort in your processing function directly, so you don't need mutability.